interface/ceed-operator.c

```diff
@@ -872,6 +872,48 @@ int CeedOperatorGetFields(CeedOperator op, CeedInt *num_input_fields, CeedOperat
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Get a CeedOperatorField of an CeedOperator from its name
+
+  Note: Calling this function asserts that setup is complete and sets the CeedOperator as immutable.
+
+  @param[in]  op         CeedOperator
+  @param[in]  field_name Name of desired CeedOperatorField
+  @param[out] op_field   CeedOperatorField corresponding to the name
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Advanced
+**/
+int CeedOperatorGetFieldByName(CeedOperator op, const char *field_name, CeedOperatorField *op_field) {
+  CeedInt            num_input_fields, num_output_fields;
+  CeedOperatorField *input_fields, *output_fields;
+  char              *name;
+  CeedCall(CeedOperatorGetFields(op, &num_input_fields, &input_fields, &num_output_fields, &output_fields));
+
+  for (CeedInt i = 0; i < num_input_fields; i++) {
+    CeedCall(CeedOperatorFieldGetName(input_fields[i], &name));
+    if (!strcmp(name, field_name)) {
+      *op_field = input_fields[i];
+      return CEED_ERROR_SUCCESS;
+    }
+  }
+
+  for (CeedInt i = 0; i < num_output_fields; i++) {
+    CeedCall(CeedOperatorFieldGetName(output_fields[i], &name));
+    if (!strcmp(name, field_name)) {
+      *op_field = output_fields[i];
+      return CEED_ERROR_SUCCESS;
+    }
+  }
+
+  bool has_name = op->name;
+  // LCOV_EXCL_START
```

Comment:
Hey, @jrwrigh can we update this function so that we can let the return value be `NULL` if the field isn't found? That would be very helpful for our plasticity branch in Ratel.

Reply:
Hmmm. I had though about returning a `NULL` in case it isn't found, but thought that it would just be more work to handle it on the user's end, at least to get a reasonable error message out of it. But I can imagine returning `NULL` being quite handy.

tl;dr Go for it.

Reply:
Maybe a separate `CeedOperator{Has,Check}Field` function could also work here.

Reply:
I think either is fine - its the same basic logic either way, just a question of one function call or two

Reply:
I kinda like `CeedOperator{Has,Check}FieldName` a bit more, mainly for keeping the easy-to-read error message.

Reply:
I just updated this in Ratel, I use the NULL return there and now use PetscCheck(theThing, ratel->comm, PETSC_ERR_SUP, "the thing not found") or if (theThing) as appropriate

Reply:
Though I think I might not need this with my new plan

Reply:
Still need to get back to the issue from this PR about checking for non-dupe fields

Reply:
Gotcha. Forgot about `PetscCheck`. That's simple enough.